dockerfiles/csvapi/Dockerfile.csvapi.with.source

```diff
@@ -4,6 +4,9 @@ WORKDIR /app
 
 COPY ./csvapi/ .
 
+RUN pip install --upgrade pip
+# hopefully caches the requirements install layer when upgrading source
```

Comment:
If `csvapi` has changes, docker won't use the cache layers for this step, since it happens after `COPY ./csvapi/`.
We could pre-copy requirements only, install, and then copy all.